v2/controllers/egress_controller.go

```diff
@@ -366,6 +373,54 @@ func (r *EgressReconciler) reconcileService(ctx context.Context, log logr.Logger
 	return nil
 }
 
+func (r *EgressReconciler) reconcilePDB(ctx context.Context, log logr.Logger, eg *coilv2.Egress) error {
+	if eg.Spec.PodDisruptionBudget == nil {
+		return nil
+	}
+
+	pdb := &policyv1.PodDisruptionBudget{}
+	pdb.Namespace = eg.Namespace
+	pdb.Name = eg.Name + "-pdb"
+
+	result, err := ctrl.CreateOrUpdate(ctx, r.Client, pdb, func() error {
```

Comment:
I suspect it would be a reasonable limitation that the PDB should be created by Coil.
This section is well-aligned with the logic of other functions, so I suspect there would be no need to change this.

Reply:
We decided to use `CreateOrUpdate()` this time for the following reasons:
- Other resources (e.g. `Service`, `Deployment`) also uses `CreateOrUpdate` in coil.
- PDB doesn't have so many fields.

It may be good to rewrite entirely by SSA as a refactoring.